src/main/ngapp/accio/src/app/editor/editor.component.ts

```diff
@@ -761,6 +761,42 @@ export class EditorComponent implements OnInit {
     this.disableSubmit = this.disableFloodFill = false;
   }
 
+  getFloodfillSet(coord: Coordinate) {
+    const maskSet = this.magicWandService.floodfill(
+      this.originalImageData,
+      coord.x,
+      coord.y, 
+      this.tolerance
+    )
+    this.floodfillMask(
+      new MaskAction(          
+        this.maskTool == MaskTool.MAGIC_WAND_ADD
+            ? Action.ADD
+            : Action.SUBTRACT,
+          Tool.MAGIC_WAND,
+          maskSet
+    ));
+  }
+
+  async getScribbleSet(coord: Coordinate): Promise<void> {
```

Comment:
This function specifies a return type but never returns anything. Is that intentional?

Reply:
the function has to wait for the painted set to be loaded before the rest of the function runs, so it has to be async, and returns void.